GeoLib/Grid.h

```diff
@@ -175,25 +177,25 @@ class Grid : public GeoLib::AABB<POINT>
 
 template <typename POINT>
 template <typename InputIterator>
-Grid<POINT>::Grid(InputIterator first, InputIterator last, std::size_t max_num_per_grid_cell) :
-	GeoLib::AABB<POINT>(first, last),
-	_n_steps({{1,1,1}}),
-	_step_sizes({{0.0,0.0,0.0}}),
-	_inverse_step_sizes({{0.0,0.0,0.0}}),
-	_grid_cell_nodes_map(nullptr)
+Grid<POINT>::Grid(InputIterator first, InputIterator last,
+	std::size_t max_num_per_grid_cell)
+	: GeoLib::AABB<POINT>(first, last), _n_steps({{1,1,1}}),
+		_step_sizes({{0.0,0.0,0.0}}), _inverse_step_sizes({{0.0,0.0,0.0}}),
+		_grid_cell_nodes_map(nullptr)
 {
 	std::size_t n_pnts(std::distance(first, last));
```

Comment:
A signed type would be safer here due to possible underflow.
From http://en.cppreference.com/w/cpp/iterator/distance
> Return value
> The number of increments needed to go from first to last. **The value may be negative** if random-access iterators are used and first is reachable from last (since C++11)